src/main/java/org/broadinstitute/hellbender/tools/walkers/vqsr/VariantRecalibrator.java

```diff
@@ -0,0 +1,873 @@
+            final boolean isInput,
+            final FeatureContext context ) {
+        if( aggregateInputs == null ) { throw new IllegalArgumentException("aggregateInputs cannot be null."); }
+        if( context == null ) { throw new IllegalArgumentException("context cannot be null."); }
+
+        for( final VariantContext vc : context.getValues(aggregateInputs, context.getInterval().getStart()) ) {
+            addVariantDatum(vc, isInput, context);
+        }
+    }
+
+    private void addVariantDatum( VariantContext vc, final boolean isInput, final FeatureContext context ) {
+        if( vc != null && ( IGNORE_ALL_FILTERS || vc.isNotFiltered() || ignoreInputFilterSet.containsAll(vc.getFilters()) ) ) {
+            if( VariantDataManager.checkVariationClass( vc, VRAC.MODE ) && !VRAC.useASannotations) {
+                addDatum(reduceSum, isInput, context, vc, null, null);
+            }
+            else if( VRAC.useASannotations ) {
+                for (final Allele allele : vc.getAlternateAlleles()) {
+                    if ( allele == Allele.SPAN_DEL ) {
+                        continue;
+                    }
+                    if ( VariantDataManager.checkVariationClass(vc, allele, VRAC.MODE )) {
+                        addDatum(reduceSum, isInput, context, vc, vc.getReference(), allele);
+                    }
+                }
+            }
+        }
+    }
+
+    /**
+     * add a datum representing a variant site (or allele) to the data in {@code variants}, which represents the callset to be recalibrated
+     * @param variants is modified by having a new VariantDatum added to it
+     */
+    private void addDatum(
+            final ExpandingArrayList<VariantDatum> variants,
+            final boolean isInput,
+            final FeatureContext featureContext,
+            final VariantContext vc,
+            final Allele refAllele,
+            final Allele altAllele) {
+        final VariantDatum datum = new VariantDatum();
+
+        // Populate the datum with lots of fields from the VariantContext, unfortunately the VC is too big so we just
+        // pull in only the things we absolutely need.
+        datum.referenceAllele = refAllele;
+        datum.alternateAllele = altAllele;
+        dataManager.decodeAnnotations(datum, vc, true);
+
+        // non-deterministic because order of calls depends on load of machine
+        datum.loc = (isInput ? new SimpleInterval(vc) : null);
+
+        datum.originalQual = vc.getPhredScaledQual();
+        datum.isSNP = vc.isSNP() && vc.isBiallelic();
+        datum.isTransition = datum.isSNP && GATKVariantContextUtils.isTransition(vc);
+        datum.isAggregate = !isInput;
+
+        // Loop through the training data sets and if they overlap this locus (and allele, if applicable) then update
+        // the prior and training status appropriately
+        dataManager.parseTrainingSets(featureContext, vc, datum, TRUST_ALL_POLYMORPHIC);
```

Comment:
Add a comment that `parseTrainingSets` is only using the start position of the `featureContext`